plugins/services/src/js/service-configuration/ServiceNetworkingConfigSection.js

```diff
@@ -114,7 +114,9 @@ class ServiceNetworkingConfigSection extends ServiceConfigBaseSectionDisplay {
                 prop: keys.port,
                 className: getColumnClassNameFn(),
                 render(prop, row) {
-                  return getDisplayValue(row[prop]);
+                  return appDefinition.requirePorts
+                    ? getDisplayValue(row[prop])
+                    : <em>Auto Assigned</em>;
```

Comment:
@ashenden I've sent you a message with an image snapshot of this.

<img width="968" alt="screen shot 2017-07-17 at 10 17 02 am" src="https://user-images.githubusercontent.com/180432/28280810-d3c6dbc0-6ad9-11e7-958c-91da237888d6.png">